File: jari_core/jari_core/doctype_backup_2026-06-22_1912/spindal_receive/spindal_receive.js
frappe.ui.form.on('Spindal Receive', {
    spindal_issue(frm) {
        if (!frm.doc.spindal_issue) return;

        frappe.call({
            method: "jari_core.jari_core.doctype.spindal_receive.spindal_receive.get_spindal_peti_entries",
            args: {
                spindal_issue: frm.doc.spindal_issue
            },
            callback(r) {
                if (!r.message) return;

                frm.clear_table('received_peti_items');

                let total_net_weight = 0;

                r.message.forEach(row => {
                    let child = frm.add_child('received_peti_items');

                    child.spindal_peti_entry = row.name;
                    child.peti_no = row.name;
                    child.batch_no = row.batch_no;
                    child.quality_code = row.quality_code;
                    child.khata_no = row.khata_no;
                    child.gross_weight = row.gross_weight;
                    child.baad_weight = row.baad_weight;
                    child.net_weight = row.net_weight;

                    total_net_weight += flt(row.net_weight);
                });

                frm.refresh_field('received_peti_items');

                if (frm.fields_dict.total_net_weight) {
                    frm.set_value('total_net_weight', total_net_weight);
                }

                if (frm.fields_dict.total_receive_weight) {
                    frm.set_value('total_receive_weight', total_net_weight);
                }
            }
        });
    }
});

File: jari_core/jari_core/doctype_backup_2026-06-22_1912/spindal_receive/spindal_receive.py
import frappe
from frappe.model.document import Document
from frappe.utils import flt


class SpindalReceive(Document):

    def validate(self):
        self.fetch_peti_entries()
        self.calculate_totals()

    def fetch_peti_entries(self):
        if not self.spindal_issue:
            return

        if self.received_peti_items:
            return

        petis = get_spindal_peti_entries(self.spindal_issue)

        for peti in petis:
            row = self.append("received_peti_items", {})
            row.spindal_peti_entry = peti.get("name")
            row.peti_no = peti.get("name")
            row.batch_no = peti.get("batch_no")
            row.quality_code = peti.get("quality_code")
            row.khata_no = peti.get("khata_no")
            row.gross_weight = flt(peti.get("gross_weight"))
            row.baad_weight = flt(peti.get("baad_weight"))
            row.net_weight = flt(peti.get("net_weight"))

    def calculate_totals(self):
        total_net = 0

        for row in self.received_peti_items or []:
            total_net += flt(row.net_weight)

        if hasattr(self, "total_net_weight"):
            self.total_net_weight = total_net

        if hasattr(self, "total_receive_weight"):
            self.total_receive_weight = total_net


@frappe.whitelist()
def get_spindal_peti_entries(spindal_issue):
    issue = frappe.get_doc("Spindal Issue", spindal_issue)

    return frappe.get_all(
        "Spindal Peti Entry",
        filters={
            "spindal_issue": spindal_issue,
            "batch_no": issue.active_batch_no,
            "docstatus": 1
        },
        fields=[
            "name",
            "spindal_issue",
            "batch_no",
            "quality_code",
            "khata_no",
            "gross_weight",
            "baad_weight",
            "net_weight"
        ],
        order_by="creation asc"
    )
